### PR TITLE
Upgraded CSI spec to v1.2.0

### PR DESCRIPTION
**Pull request by jqmichael**
**What is this PR about? / Why do we need it?**
Upgraded CSI Spec to the latest. None of the change applies to EFS code base. 

```
Breaking changes/Deprecations:

None (CSI 1.x releases are backwards compatible).
Additions:

spec (PR #372): Add optional secrets field to ListSnapshotsRequest.
spec (PR #374): Add optional (but recommended) VolumeStatus.published_node_ids field to ListVolumesResponse (to enable COs to detect and reconcile when a volume is unpublished out of band).
spec (PR #390): Add optional (but recommended) staging_path to NodeGetVolumeStatsRequest and NodeExpandVolumeRequest.
spec (PR #381): Add optional volume_capability to ControllerExpandVolumeRequest and NodeExpandVolumeRequest (to enable SPs to differentiate between block and file).
Clarifications/Corrections/Fixes:

spec (PR #347): Clarify DeleteVolume behavior with snapshots.
spec (PR #375): Clarify SP is allowed to return OK for ControllerUnpublishResponse if node or volume not found.
spec (PR #383): Relax recovery behavior for NOT_FOUND error code for ControllerUnpublishVolume from MUST to SHOULD.
Dependencies:

lib/go (PR #378) Update protoc to v3.9.1, protoc-gen-go to v1.3.2.
```
https://github.com/container-storage-interface/spec/releases

**What testing is done?** 
Tested manually.

### CODE REVIEW

File: deploy/kubernetes/base/node.yaml [hunk 1]
```diff
@@ -17,7 +17,6 @@ spec:
       nodeSelector:
         kubernetes.io/os: linux
         kubernetes.io/arch: amd64
-      hostNetwork: true
```
Comment (leakingtapan) [inline deploy/kubernetes/base/node.yaml:20]: Please update the helm chart as well to make them consistent: https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/helm

File: deploy/kubernetes/base/node.yaml [hunk 2]
```diff
@@ -26,7 +25,6 @@ spec:
           securityContext:
             privileged: true
           image: amazon/aws-efs-csi-driver:latest
-          imagePullPolicy: Always
```
Comment (leakingtapan) [inline deploy/kubernetes/base/node.yaml:29]: Same above

File: deploy/kubernetes/base/node.yaml [hunk 3]
```diff
@@ -56,7 +54,6 @@ spec:
             failureThreshold: 5
         - name: csi-driver-registrar
           image: quay.io/k8scsi/csi-node-driver-registrar:v1.3.0
-          imagePullPolicy: Always
```
Comment (leakingtapan) [inline deploy/kubernetes/base/node.yaml:59]: Same above

File: deploy/kubernetes/base/node.yaml [hunk 4]
```diff
@@ -76,7 +73,6 @@ spec:
             - name: registration-dir
               mountPath: /registration
         - name: liveness-probe
-          imagePullPolicy: Always
```
Comment (leakingtapan) [inline deploy/kubernetes/base/node.yaml:79]: Same

**Comment by jqmichael** [comment]
> Is there any new APIs from 1.2.0 spec that recommend to be implemented? I don't see there is any new functionalities implemented along with the new spec.
> 
> And please update the README also

There's no new API introduced. All changes from v1.2.0 are listed in the PR description. There's none that applies to EKS CSI Driver today.

Will update the README.

File: go.mod [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-	github.com/onsi/gomega v1.5.0
-	google.golang.org/grpc v1.23.0
+	github.com/aws/aws-k8s-tester/e2e/tester v0.0.0-20200527022539-8820bddbc639 // indirect
+	github.com/aws/aws-sdk-go v1.31.2
+	github.com/container-storage-interface/spec v1.2.0
+	github.com/golang/mock v1.4.3
+	github.com/kubernetes-csi/csi-test v2.2.0+incompatible
+	github.com/onsi/ginkgo v1.12.2
+	github.com/onsi/gomega v1.10.1
+	google.golang.org/grpc v1.26.0
```
Comment (jqmichael) [inline go.mod:11]: Had to use an version prior to v1.27.0 until we pick up etcd-io/etcd#11564 (had trouble picking that commit up with go mod)

File: deploy/kubernetes/base/node.yaml (hunk 1 above)
Comment (jqmichael) [reply deploy/kubernetes/base/node.yaml]: Actually similar change has been picked up in https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/159. 

Merged from master.

File: deploy/kubernetes/base/node.yaml (hunk 2 above)
Comment (jqmichael) [reply deploy/kubernetes/base/node.yaml:28]: imagePullPolicy was not added in helm (so no need to change).

File: deploy/kubernetes/base/node.yaml (hunk 3 above)
Comment (jqmichael) [reply deploy/kubernetes/base/node.yaml:58]: imagePullPolicy was not added in helm (so no need to change).

File: deploy/kubernetes/base/node.yaml (hunk 4 above)
Comment (jqmichael) [reply deploy/kubernetes/base/node.yaml:78]: imagePullPolicy was not added in helm (so no need to change).

**Comment by jqmichael** [comment]
@leakingtapan Please take a look. Thanks

File: docs/README.md [hunk 6]
```diff
@@ -6,13 +6,13 @@
...
-| AWS EFS CSI Driver \ CSI Version       | v0.3.0| v1.1.0 |
-|----------------------------------------|-------|--------|
-| master branch                          | no    | yes    |
-| v0.3.0                                 | no    | yes    |
-| v0.2.0                                 | no    | yes    |
-| v0.1.0                                 | yes   | no     |
+### CSI Specification Compatibility Matrix
+| AWS EFS CSI Driver \ CSI Spec Version  | v0.3.0| v1.1.0 | v1.2.0 |
+|----------------------------------------|-------|--------|--------|
+| master branch                          | no    | yes    | yes    |
```
Comment (leakingtapan) [inline docs/README.md:12]: v1.1.0 should be no for master branch

Comment (jqmichael) [reply docs/README.md]: Nice catch. Fixed

**Pull request merged**